File: source/plugin/Assets/Adivery/RewardedCallback.cs
using UnityEngine;

namespace AdiveryUnity
{
    internal class RewardedCallback : AndroidJavaProxy
    {
        public RewardedCallback() : base("com.adivery.unity.RewardedCallback") { }

        public virtual void onAdLoaded() { }

        public virtual void onAdRewarded() { }

        public virtual void onAdLoadFailed() { }

        public virtual void onAdShown() { }

        public virtual void onAdShowFailed() { }

        public virtual void onAdClicked() { }

        public virtual void onAdClosed() { }
    }
}

File: source/android-library/app/src/main/java/com/adivery/unity/RewardedCallback.java
package com.adivery.unity;

public interface RewardedCallback {
  void onAdLoaded();

  void onAdLoadFailed(int errorCode);

  void onAdShown();

  void onAdShowFailed(int errorCode);

  void onAdRewarded();

  void onAdClicked();

  void onAdClosed();
}
